src/components/CoverageCard/CoverageCard.tsx

```diff
@@ -0,0 +1,166 @@
+import type { Contact, CoverageEntry } from '@prezly/sdk';
+import classNames from 'classnames';
+import type { ReactNode } from 'react';
+
+import { formatBytes } from '../../lib';
+
+import { getCoverageImageUrl } from './getCoverageImageUrl';
+
+interface Props {
+    className?: string;
+    coverage: CoverageEntry;
+    layout: 'vertical' | 'horizontal';
+    newTab: boolean;
+    renderDate: (date: string) => ReactNode;
+    withThumbnail: boolean;
+}
+
+export function CoverageCard({
+    className,
+    coverage,
+    layout,
+    newTab,
+    renderDate,
+    withThumbnail,
+}: Props) {
+    const imageUrl = getCoverageImageUrl(coverage);
+    const href = coverage.attachment_oembed?.url || coverage.url;
+    const autoLayout = withThumbnail && imageUrl ? layout : 'horizontal';
+
+    return (
+        <div
+            className={classNames('prezly-slate-coverage-card-component__container', className, {
+                horizontal: autoLayout === 'horizontal',
+                vertical: autoLayout === 'vertical',
+            })}
+        >
+            {imageUrl && withThumbnail && <Thumbnail src={imageUrl} href={href} newTab={newTab} />}
+
+            <div className="prezly-slate-coverage-card-component__details">
+                <Title coverage={coverage} href={href} newTab={newTab} />
+
+                <Description coverage={coverage} />
+
+                <Meta
+                    author={coverage.author_contact}
+                    date={coverage.published_at}
+                    outlet={coverage.organisation_contact}
+                    renderDate={renderDate}
+                />
+            </div>
+        </div>
+    );
+}
+
+function Thumbnail(props: { href: string | null; newTab: boolean; src: string }) {
```

Comment:
Should we extract these components into separate files and export, so that people can re-use them when implementing custom Coverage node component?

Reply:
I don't think we need that at this point. We haven't been doing that with previous components either and no one has complained, so.. YAGNI 😁